backend/src/main/java/com/example/backend/service/CollectionsService.java

```diff
@@ -0,0 +1,45 @@
+package com.example.backend.service;
+
+
+import com.example.backend.model.MetadataFromCollection;
+import com.example.backend.model.NFTCollectionsResponseElement;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+
+@Service
+public class CollectionsService {
+
+    private final WebClient client = WebClient.create("https://api.nftport.xyz/v0/");
+
+    private MetadataFromCollection AllNfts() {return AllNfts();}
+
+
+
+     public List<NFTCollectionsResponseElement> getNftsByQuery(String query){
+         NFTCollectionsResponseElement responseElement =
+                 client.get()
+                         .uri("nfts?chain=ethereum&include=metadata&Authorization=d4f5003c-bf80-4a38-8b65-444e985d8919" + query)
+                         .retrieve()
+                         .toEntity(NFTCollectionsResponseElement.class)
+                         .block()
+                         .getBody();
+         return responseElement.getResult();
+     }
+
+    public MetadataFromCollection getAllNfts() {
+        MetadataFromCollection AllNfts =
+                client.get()
+                        .uri("nfts?chain=ethereum&include=metadata&Authorization=d4f5003c-bf80-4a38-8b65-444e985d8919")
```

Comment:
Den API Key solltest du noch in einem zweiten Schritt auslagern, damit er nicht im Github Repo liegt :)